_includes/markdown/Typescript.md

```diff
@@ -19,7 +19,7 @@ The main reason we don't require TypeScript for standard WordPress projects at t
 
 ## How to use TypeScript
 
-In an ideal world, every engineer would be fully capable of writing all sorts of complex TypeScript types, however, we understand that it takes time to master a language like TypeScript, especially in a few scenarios that require writing more complex types. Therefore, while we use a reasonably strict `tsconfig.json`` we are flexible in practice. 
+In an ideal world, every engineer would be fully capable of writing all sorts of complex TypeScript types, however, we understand that it takes time to master a language like TypeScript, especially in a few scenarios that require writing more complex types. Therefore, while we use a reasonably strict `tsconfig.json` we are flexible in practice.
```

Comment:
@nicholasio I would argue that this:
```
every engineer would be fully capable of writing all sorts of complex TypeScript types
```
is something you'd ideally want to avoid 😅 While TypeScript can be absolutely amazing as to what it can achieve (see Prisma dynamically generated DB types for example) - we feel like this adds an immense complexity overhead that is not needed. It let's you do things that no statically typed language lets you do - and this can add a lot of complexity and confusion to code. Obviously, there are times where it's super helpful, but in most cases it's likely to take up too much time for something that could be done simpler - and this is what, I think, you should try and promote - using TS, but not going overboard with it. Basically - use it like you would with Java, .NEt or similar types.

I'm well aware I'm not providing an alternative for the particular highlighted text, instead I am trying to provide some insight as to how far you may want to go with TS, and at what point would you may want to stop 😅

Reply:
@kasparszarinovs Do you have an example of going overboard with TS? It is not something I feel like I have ran into on large and complex TS projects. So for my benefit I'd love to understand more about this and perhaps I'm going overboard and not realising it and then making things more complicated for other engineers

Reply:
@tobeycodes I do, but I'll have to manually re-create the situation as we normally reason them as part of PR process 😅 I'll send something over during next week if that works!

Reply:
@kasparszarinovs I see where you're coming from, definately not the intention I'll rewrite this to better convey the message that "ideally no one would struggle with TS"

Reply:
I think the issue here is that implementing TypeScript in certain circumstances can force engineers down a rabbit hole, especially when combining intricate types across technologies or implementations - for instance, `ReactElement` and the `HTMLButtonElement`. 

I've experienced this myself on projects where getting types right becomes a bottleneck, and the value begins to slip. Unfortunately, I think this essentially boils down to experience—these situations aren't frequent—they're more like edge cases, which can be solved mainly by reaching out to more experienced team members or looking at code. This should be communicated to team members who need more experience in the language.

That said, like with most technologies, we should anticipate a slow ramp up for most engineers unfamiliar with TypeScript and its conventions

Reply:
@dainemawer What you're talking about is the cost of ramping up engineers to TypeScript to which we hope to minimize with the tone and expectations that we're setting up here alongside the escape hatches we are strategically allowing.

The other side of the coin which we hope to address as well here is overengineering a codebase with types that are more complex than it needs to be for the limited purposes of projects.

Reply:
@dainemawer Perhaps this document could give some examples on how to solve common problems like React onClick event handling. That way engineers can check that list rather than going down the rabbit hole of Google (which will give you ton of mixed results) or asking another engineer

Reply:
Added a paragraph about https://react-typescript-cheatsheet.netlify.app/ I think that will suffice for now. We can interate and document the things we see people struggling the most in the future.

Reply:
Also updated the wording here to avoid overengineering types.

Reply:
Never knew that cheat sheet existed, so great idea to add

Reply:
Agreed, thanks for the insight @nicholasio @tobeycodes - I think some top of mind solutions to challenges can even come after this Best Practice doc is released. Fabian and I started a Git Discussions for FSE which could be helpful for TypeScript as well: https://github.com/10up/gutenberg-best-practices/discussions - the team has already started logging questions there for Gutenberg, may be a nice solution for TypeScript.